test/parallel/test-assert.js

```diff
@@ -559,6 +560,16 @@ a.throws(makeBlock(thrower, TypeError), (err) => {
       code: 'ERR_ASSERTION',
       message: /^Missing expected exception \(TypeError\): fhqwhgads$/
     }));
+
+  let threw = false;
+  try {
+    assert.throws(noop);
+  } catch (e) {
+    threw = true;
+    assert.ok(e instanceof a.AssertionError);
+    assert.ok(!e.stack.includes('at Function.throws'));
```

Comment:
As requested, I've restored the previous test, but I cannot extend an existing test because:
- Only 4 tests a triggering the "Missing expected exception", which I'm looking for
- those 4 tests are using `common.expectsError()`, and a a `assert.throws(() => { a.throws(` pattern which makes the stack trace more complicated than it should.

Without stack trace cleaning:
```
AssertionError [ERR_ASSERTION]: Missing expected exception.
    at new AssertionError (internal/errors.js:320:11)
    at innerFail (assert.js:74:9)
    at expectsError (assert.js:439:5)
    at Function.throws (assert.js:476:3)
    at Object.<anonymous> (D:\Workspaces\perso\node\test\parallel\test-assert.js:566:12)
```

With cleaning:
```
AssertionError [ERR_ASSERTION]: Missing expected exception.
    at Object.<anonymous> (D:\Workspaces\perso\node\test\parallel\test-assert.js:566:12)
```